modules/shortcodes.php

```diff
@@ -118,12 +118,51 @@ function jetpack_preg_replace_callback_outside_tags( $pattern, $callback, $conte
 	
 	$textarr = wp_html_split( $content );
 	unset( $content );
+
+	// Flag to check we're inside an HTML tag.
+	$inside_tag = '';
+
 	foreach( $textarr as &$element ) {
-	    if ( '' === $element || '<' === $element{0} )
+
+		// If it's an empty element, continue.
+	    if ( '' === $element ) {
 	        continue;
-	    $element = preg_replace_callback( $pattern, $callback, $element );
+		}
+
+		// If this is an opening HTML tag, save it and move to the next element.
+		if ( '<' === $element[0] && '/>' !== substr( $element, -2 ) ) {
+			$inside_tag = $element;
+			continue;
+		}
+
+		// If this is a closing HTML tag, erase the previously saved element.
+		if ( '</' === substr( $element, 0, 2 ) ) {
+			$inside_tag = '';
+			continue;
+		}
+
+		// We're inside an HTML tag.
+		if ( ! empty( $inside_tag ) ) {
+
+			// If the callback it's not in the blacklist, run it.
```

Comment:
This seems confusing. It doesn't look like we are blacklisting or whitelisting. It looks like instead we're checking for custom rules and then running a default if none matches.